lib/Conversion/PipelineToHW/PipelineToHW.cpp

```diff
@@ -175,21 +180,59 @@ class PipelineOutlineLowering : public PipelineLowering {
     return builder.getStringAttr("s" + std::to_string(stageIdx));
   }
 
+  // Helper class to manage grabbing the various inputs for stage modules.
+  struct PipelineStageMod {
+    PipelineStageMod(PipelineOutlineLowering &parent, Block *block,
+                     hw::HWModuleOp mod) {
+      size_t nStageInputArgs = parent.pipeline.getNumStageArguments(block);
+      inputs = mod.getArguments().take_front(nStageInputArgs);
+      extInputs = mod.getArguments().slice(
+          nStageInputArgs, parent.getStageExtInputs(block).size());
+      clock = mod.getArgument(mod.getNumArguments() - 2);
```

Comment:
Magic constants. Eww. Can you define some (perhaps public) offset constants for these? Or an attribute? We should really come up with an efficient way to access port indexes by name.